contracts/payment/PullPayment.sol

```diff
@@ -10,34 +9,34 @@ import "../math/SafeMath.sol";
  * contract and use asyncSend instead of send or transfer.
  */
 contract PullPayment {
-  using SafeMath for uint256;
+  Escrow public escrow;
 
-  mapping(address => uint256) public payments;
-  uint256 public totalPayments;
+  constructor() public {
+    escrow = new Escrow();
+  }
 
   /**
   * @dev Withdraw accumulated balance, called by payee.
   */
   function withdrawPayments() public {
     address payee = msg.sender;
-    uint256 payment = payments[payee];
-
-    require(payment != 0);
-    require(address(this).balance >= payment);
-
-    totalPayments = totalPayments.sub(payment);
-    payments[payee] = 0;
+    escrow.withdraw(payee);
+  }
 
-    payee.transfer(payment);
+  /**
+  * @dev Returns the credit owed to an address.
+  & @param _dest The creditor's address.
+  */
+  function payments(address _dest) public view returns (uint256) {
+    return escrow.depositsOf(_dest);
   }
 
   /**
   * @dev Called by the payer to store the sent amount as credit to be pulled.
-  * @param dest The destination address of the funds.
-  * @param amount The amount to transfer.
+  * @param _dest The destination address of the funds.
+  * @param _amount The amount to transfer.
   */
-  function asyncSend(address dest, uint256 amount) internal {
-    payments[dest] = payments[dest].add(amount);
-    totalPayments = totalPayments.add(amount);
+  function asyncSend(address _dest, uint256 _amount) internal {
```

Comment:
The word `transfer` is quite overloaded but I feel like this is an `asyncTransfer` more than an `asyncSend`, in analogy with `send` vs `transfer` because of the lack of return value.

Reply:
I missed this when updating `PullPayment`, I agree that `transfer` is indeed the better term.